Files/UserControls/SearchBox.xaml.cs

```diff
@@ -7,27 +12,52 @@ namespace Files.UserControls
 {
     public sealed partial class SearchBox : UserControl
     {
-        public SearchBoxViewModel SearchBoxViewModel
-        {
-            get => (SearchBoxViewModel)GetValue(SearchBoxViewModelProperty);
-            set => SetValue(SearchBoxViewModelProperty, value);
-        }
+        private readonly SearchNavigator navigator = new SearchNavigator();
 
-        // Using a DependencyProperty as the backing store for SearchBoxViewModel.  This enables animation, styling, binding, etc...
         public static readonly DependencyProperty SearchBoxViewModelProperty =
             DependencyProperty.Register(nameof(SearchBoxViewModel), typeof(SearchBoxViewModel), typeof(SearchBox), new PropertyMetadata(null));
 
-        public SearchBox()
+        public SearchBoxViewModel SearchBoxViewModel
         {
-            InitializeComponent();
+            get => (SearchBoxViewModel)GetValue(SearchBoxViewModelProperty);
+            set => SetValue(SearchBoxViewModelProperty, value);
         }
 
-        private void SearchRegion_TextChanged(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs e) => SearchBoxViewModel.SearchRegion_TextChanged(sender, e);
+        public SearchBox() => InitializeComponent();
 
-        private void SearchRegion_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs e) => SearchBoxViewModel.SearchRegion_QuerySubmitted(sender, e);
+        private void SearchRegion_TextChanged(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs e)
+            => SearchBoxViewModel.SearchRegion_TextChanged(sender, e);
+        private void SearchRegion_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs e)
+            => SearchBoxViewModel.SearchRegion_QuerySubmitted(sender, e);
+        private void SearchRegion_SuggestionChosen(AutoSuggestBox sender, AutoSuggestBoxSuggestionChosenEventArgs e)
+            => SearchBoxViewModel.SearchRegion_SuggestionChosen(sender, e);
+        private void SearchRegion_Escaped(KeyboardAccelerator sender, KeyboardAcceleratorInvokedEventArgs e)
+            => SearchBoxViewModel.SearchRegion_Escaped(sender, e);
 
-        private void SearchRegion_SuggestionChosen(AutoSuggestBox sender, AutoSuggestBoxSuggestionChosenEventArgs e) => SearchBoxViewModel.SearchRegion_SuggestionChosen(sender, e);
 
-        private void SearchRegion_Escaped(KeyboardAccelerator sender, KeyboardAcceleratorInvokedEventArgs e) => SearchBoxViewModel.SearchRegion_Escaped(sender, e);
+        private void MenuFrame_Loaded(object sender, RoutedEventArgs e)
+        {
+            navigator.SearchBox = SearchBoxViewModel;
+            navigator.Frame = sender as Frame;
+            GoRootPage();
+        }
+        private void MenuButton_Loaded(object sender, RoutedEventArgs e)
+        {
+            var allowFocusOnInteractionAvailable = ApiInformation.IsPropertyPresent("Windows.UI.Xaml.FrameworkElement", "AllowFocusOnInteraction");
```

Comment:
Should this be static so we don't check it every time?

Reply:
@d2dyno1 This is what I would have done for another control. But this is always displayed, and therefore this event is only triggered once. It is useless to be encumbered with a static field which will not be reused.